pkgs/development/interpreters/lua-5/interpreter.nix

```diff
@@ -42,34 +42,50 @@ stdenv.mkDerivation (finalAttrs:
   {
   pname = "lua";
   inherit version;
+
   outputs = [ "out" "doc" ];
 
   src = fetchurl {
-    url = "https://www.lua.org/ftp/${finalAttrs.pname}-${finalAttrs.version}.tar.gz";
+    url = "https://www.lua.org/ftp/lua-${finalAttrs.version}.tar.gz";
     sha256 = hash;
   };
 
   LuaPathSearchPaths  = luaPackages.luaLib.luaPathList;
   LuaCPathSearchPaths = luaPackages.luaLib.luaCPathList;
+
+  # hook
+  LUA_PATH_PATTERN = [
+    "share/lua/${luaversion}/?.lua"
+    "share/lua/${luaversion}/?/init.lua"
+  ]
+  ++ lib.optionals (lib.versionAtLeast self.luaversion "5.3") [
+    "./?.lua"
+    "./?/init.lua"
+    "lib/lua/${luaversion}/?.lua"
+    "lib/lua/${luaversion}/?/init.lua"
+  ];
```

Comment:
1. now that we're not overriding LUA_PATH_DEFAULT, we don't need to handle relative module paths in nix, since we aren't overriding the part of luaconf.h that already handles this
2. this is repeating the mistake i mentioned in #293832
3. this is wrong, lua 5.1 also has `./?.lua` in its path, just not `./?/init.lua`